### PR TITLE
Add syntax for poly structs in require/typed

### PR DESCRIPTION
**Pull request by takikawa**
This adds syntax for supporting type variables in `require/typed`. Unfortunately due to contract generation issues, this doesn't actually work in normal Typed Racket.

Using unsafe imports, however, makes it useful. So some design questions: (1) should I make it only work in `unsafe-require/typed` and spit out an error in TR? (e.g., "polymorphic structs in require/typed are not supported") (2) the interaction with inheritance seems to be wonky, maybe I should only allow it for no-parent structs?

### CODE REVIEW

File: typed-racket-lib/typed-racket/base-env/prims-contract.rkt [hunk 1]
```diff
@@ -152,7 +153,8 @@
    (pattern oc:opaque-clause #:attr spec
      #`(require/opaque-type oc.ty oc.pred #,lib . oc.opt))
    (pattern (~var strc (struct-clause legacy)) #:attr spec
-     #`(require-typed-struct strc.nm (strc.body ...) strc.constructor-parts ...
+     #`(require-typed-struct strc.nm (strc.tvar ...)
```
Comment (stamourv) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: Doesn't this make the list of type variables mandatory?

Comment (takikawa) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: There is a default for this syntax class attribute (`strc.tvar`) that is the empty list.

Comment (stamourv) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: Oh, right. I missed it. Nevermind.

**Comment by samth** [comment]
@takikawa is this done?

**Comment by takikawa** [comment]
@samth It's done in the sense that it's implemented and tested, but the two design questions in the original comment remains. That is, (1) should we still merge it even if it doesn't work in normal TR? And should it be (2) disallowed on structs with parents? (my opinion is yes to both since that's still conservative)

**Comment by samth** [comment]
I agree on yes to both.

**Comment by takikawa** [comment]
Ok I plan to merge this soon. I couldn't reproduce the issues with struct inheritance I saw earlier, so I added a test for it. I also made it raise an error when type variables are provided and it's not in unsafe mode.

**Comment by takikawa** [comment]
Merged.

**Pull request closed without merging**